pkg/decision/evaluator/matchers/exact.go

```diff
@@ -27,41 +27,42 @@ import (
 )
 
 // ExactMatcher matches against the "exact" match type
-func ExactMatcher(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer, reasons decide.DecisionReasons) (bool, error) {
+func ExactMatcher(condition entities.Condition, user entities.UserContext, logger logging.OptimizelyLogProducer) (bool, decide.DecisionReasons, error) {
+	reasons := decide.NewDecisionReasons(nil)
```

Comment:
Looks like we do not use "reasons" in most matchers and conditions, just adding a lot of boilerplate codes. Can we remove "reasons" from them.
I understand one case above we have meaning messages. We can consider to use "err" message (or if it's not easy, we can just drop from reasons - they can use debug message why this happened).